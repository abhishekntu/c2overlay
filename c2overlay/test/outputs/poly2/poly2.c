/****************************************************************/
/*   poly2		                                        */
/****************************************************************/

#ifdef TEST
#include <stdio.h>
#endif

#include<math.h>

#ifdef ALGO
int foo(int x,int y)
{	
int t1_1,t1_2,t1_3,t1_4,t1_5,t1_6,t1_7,t1_8,t1_9,t1_10,t2_1,t2_2,t2_3,t2_4,t2_5,t2_6,t2_7,t2_8,t3_1,t3_2,t3_3,t3_4,t3_5,t3_6,t3_7,t4_1,t4_2,t4_3,t4_4,t5_1,t5_2,t5_3,t6_1,t6_2,t6_3,t7_1,t7_2,t8_1,t8_2,t9_1,t10_1,t11_1;

t1_1 = y - 1;
t2_1 = x * x;
t2_2 = t1_1 * y;
t3_1 = 3 * y;
t4_1 = 2*t2_1 - t3_1;
t4_2 = t2_2 * t2_2;
t5_1 = t2_1 * t4_1;
return(t5_1 + t4_2);

}
#endif


#ifdef TEST
int main(void)
{
	int x,y;
 	int result;
  	result = foo(x,y);
  	return 0;
}
#endif
