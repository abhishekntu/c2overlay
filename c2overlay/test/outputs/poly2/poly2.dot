digraph graphname {
N10 [color="black", label="O0_N12_N10"];
N8 [color="black", label="mul_N5_N8"];
N9 [color="black", label="mul_Imm_3_N6_N9"];
N1 [color="black", label="I0_N1_N1"];
N2 [color="black", label="I1_N2_N2"];
N3 [color="black", label="sqr_N11_N3"];
N4 [color="black", label="sub_Imm_1_N8_N4"];
N5 [color="black", label="sqr_N9_N5"];
N6 [color="black", label="N3N10_mul_N3_add_N10_N6"];
N7 [color="black", label="N4N7_mul_Imm_2_N4_sub_N7_N7"];
N8 -> N3;
N9 -> N7;
N1 -> N5;
N2 -> N8;
N2 -> N9;
N2 -> N4;
N3 -> N6;
N4 -> N8;
N5 -> N7;
N5 -> N6;
N6 -> N10;
N7 -> N6;
}
