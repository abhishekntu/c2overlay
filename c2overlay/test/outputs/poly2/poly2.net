.input N1
pinlist: N1
    
.input N2
pinlist: N2
    
.output out:N10
pinlist: N10
    
.clb N8_blk
pinlist: N4 N2 open open N8 open open open open 
subblock: N8_blk 0 1 open open 4 open open open open

.clb N9_blk
pinlist: N2 open open open N9 open open open open 
subblock: N9_blk 0 open open open 4 open open open open

.clb N3_blk
pinlist: N8 open open open N3 open open open open 
subblock: N3_blk 0 open open open 4 open open open open

.clb N4_blk
pinlist: N2 open open open N4 open open open open 
subblock: N4_blk 0 open open open 4 open open open open

.clb N5_blk
pinlist: N1 open open open N5 open open open open 
subblock: N5_blk 0 open open open 4 open open open open

.clb N6_blk
pinlist: N3 N5 N7 open N10 open open open open 
subblock: N6_blk 0 1 2 open 4 open open open open
    
.clb N7_blk
pinlist: N5 N9 open open N7 open open open open 
subblock: N7_blk 0 1 open open 4 open open open open

