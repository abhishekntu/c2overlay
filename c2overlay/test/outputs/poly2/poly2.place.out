Netlist file: test.net   Architecture file: test.arch.xml
Array size: 3 x 3 logic blocks

#block name	x	y	subblk	block number
#----------	--	--	------	------------
N1		3	4	0	#0
N2		4	2	0	#1
out:N10		0	2	0	#2
N8_blk		2	1	0	#3
N9_blk		3	2	0	#4
N3_blk		1	1	0	#5
N4_blk		3	1	0	#6
N5_blk		3	3	0	#7
N6_blk		1	2	0	#8
N7_blk		2	2	0	#9
